@ndlib/gatsby-theme-marble/src/store/actions/loginActions/index.js

```diff
@@ -1,38 +1,151 @@
+import OktaAuth from '@okta/okta-auth-js'
+import {
+  userIdFromClaims,
+} from 'utils/auth'
+export const GET_AUTHENTICATION = 'GET_AUTHENTICATION'
+export const AUTHENTICATE_USER = 'AUTHENTICATE_USER'
+export const GET_USER = 'GET_USER'
+export const NO_USER = 'NO_USER'
 export const LOG_USER_IN = 'LOG_USER_IN'
 export const LOG_USER_OUT = 'LOG_USER_OUT'
 export const SET_AUTH_CLIENT = 'SET_AUTH_CLIENT'
 export const STATUS_NOT_LOGGED_IN = 'STATUS_NOT_LOGGED_IN'
+export const STATUS_TRYING_AUTHENTICATION = 'STATUS_TRYING_AUTHENTICATION'
+export const STATUS_AUTHENTICATED = 'STATUS_AUTHENTICATED'
+export const STATUS_AUTHENTICATION_FAILED = 'STATUS_AUTHENTICATION_FAILED'
+export const STATUS_AUTHENTICATED_TRYING_LOGIN = 'STATUS_AUTHENTICATED_TRYING_LOGIN'
+export const STATUS_AUTHENTICATED_NOT_LOGGED_IN = 'STATUS_AUTHENTICATED_NOT_LOGGED_IN'
 export const STATUS_LOGGED_IN = 'STATUS_LOGGED_IN'
 
-export const handleLogin = (idToken) => {
+export const putAuthSettingsInStore = (site, location) => {
   return dispatch => {
-    return dispatch(logUserIn(idToken))
+    const { url, clientId, issuer } = site.siteMetadata.authClient
+    const authClientSettings = {
+      url: url,
+      clientId: clientId,
+      redirectUri: `${location.origin}/user`,
+      issuer: issuer,
+      ignoreSignature: true,
+    }
+    const { userContentPath } = site.siteMetadata
+    dispatch(setAuthClient(authClientSettings, userContentPath))
   }
 }
 
-export const logUserIn = (idToken) => {
-  const { claims, issuer } = idToken
+export const setAuthClient = (authClientSettings, userContentPath) => {
   return {
-    type: LOG_USER_IN,
-    user: {
-      username: claims.netid,
-      name: claims.name,
-      email: claims.email,
-      issuer: issuer,
-      token: idToken,
-    },
+    type: SET_AUTH_CLIENT,
+    authClientSettings: authClientSettings,
+    userContentPath: userContentPath,
+  }
+}
+
+export const getTokenAndPutInStore = (loginReducer, location) => {
+  return dispatch => {
+    if (loginReducer.authClientSettings) {
+      const authClient = new OktaAuth(loginReducer.authClientSettings)
+      try {
+        authClient.tokenManager.get('idToken')
+          .then(idToken => {
+            if (idToken) {
+              if (loginReducer.status === 'STATUS_NOT_LOGGED_IN') {
+                dispatch(storeAuthenticationAndGetLogin(idToken, loginReducer))
+              }
+              // If ID Token isn't found, try to parse it from the current URL
+            } else if (location.hash) {
+              authClient.token.parseFromUrl()
```

Comment:
This needs a catch or else the store may get stuck in a fetching status.

Reply:
added catch